src/tools/tests/manifest2proto-test.ts

```diff
@@ -414,6 +414,106 @@ describe('manifest2proto', () => {
     });
   });
 
+  it('encodes variable type - writeSuperset constraint', async () => {
```

Comment:
I would like a test that is explicitly serializing the particle with a constrained TV, e.g.

```
particle TimeRedactor
  input: reads ~a with {time: Number}
  output: writes ~a
```